lib/rest/RequestHandler.js

```diff
@@ -299,12 +299,13 @@ class RequestHandler {
                         if(resp.statusCode >= 300) {
                             if(resp.statusCode === 429) {
                                 const content = typeof body === "object" ? `${body.content} ` : "";
-                                this._client.emit("debug", `${resp.headers["x-ratelimit-global"] ? "Global" : "Unexpected"} 429 (╯°□°）╯︵ ┻━┻: ${response}\n${content} ${now} ${route} ${resp.statusCode}: ${latency}ms (${this.latencyRef.latency}ms avg) | ${this.ratelimits[route].remaining}/${this.ratelimits[route].limit} left | Reset ${this.ratelimits[route].reset} (${this.ratelimits[route].reset - now}ms left)`);
-                                if(retryAfter) {
+                                const delay = resp.headers['retry-after'] ? Number(resp.headers['retry-after']) * 1000 : retryAfter
```

Comment:
Handle shared ratelimits by using `response.retry-after`, not from the header (keep in mind that response is a string and needs to be deserialized).

Reply:
that - should be _ but yes, or you can use x-ratelimit-reset-after

Reply:
Headers aren't correct/accurate for shared ratelimits, you'll have to use `response.retry_after`.